camel/models/base_model.py

```diff
@@ -138,3 +142,83 @@ def stream(self) -> bool:
             bool: Whether the model is in stream mode.
         """
         return False
+
+
+class ModelManager:
+    def __init__(
+        self,
+        models: Union[BaseModelBackend, List[BaseModelBackend]],
+        scheduling_strategy: str = "round_robin",
+    ):
+        """
+        Initialize the ModelManager with a list of models and
+        a scheduling strategy.
+        models: model backend or list of model backends
+        (e.g., model instances, APIs)
+        scheduling_strategy: name of function that defines how
+         to select the next model.
+        Defaults to round-robin if not provided.
+        """
+        if isinstance(models, list):
+            self.models = models
+        else:
+            self.models = [models]
+        self.models_cycle = cycle(self.models)
+        # Set the scheduling strategy; default is round-robin
+        try:
+            self.scheduling_strategy = getattr(self, scheduling_strategy)
+        except AttributeError:
+            logger.warning(
+                f"Provided strategy: {scheduling_strategy} is not implemented."
+                f"Using default 'round robin'"
+            )
+            self.scheduling_strategy = self.round_robin
+
+    @property
+    def model_type(self):
+        """Return type of the first model.
+        Assuming that all models instances are the same type."""
+        return self.models[0].model_type if self.models else None
```

Comment:
actually we shouldn't make this assumption, `model_type` could be vary , same as other properties

Reply:
Changed according to this requirement.